src/main/java/com/rsr5/gocd/riemann_notifier/PipelineDetailsPopulator.java

```diff
@@ -26,11 +41,42 @@ private JsonObject mergeInPipelineInstanceDetails(String attributeName,
         return json;
     }
 
-    protected JsonElement downloadPipelineDetails(String pipelineName) throws IOException {
-        String path = "/go/api/pipelines/" + pipelineName + "/history";
-        JsonObject json = accessor.get(path).getAsJsonObject();
-        JsonArray pipelines = json.get("pipelines").getAsJsonArray();
-        return pipelines.get(0);
+    /**
+     * Performs a depth-first search of the artifacts, returning a JsonObject
+     * translation of the first artifact named "results.xml".
+     */
+    protected JsonElement findJunitResults(JsonArray artifacts) throws IOException {
+        for (int i = 0; i < artifacts.size(); i++) {
+            JsonObject artifact = artifacts.get(i).getAsJsonObject();
+            if ("folder".equals(artifact.get("type").getAsString())) {
```

Comment:
Do you need an explicit equals method call here? Would be clearer as:
```suggestion
            if ("folder" == artifact.get("type").getAsString()) {
```

Reply:
Unfortunately yes, we do need an explicit equals call here.

Reply:
Oh, right; `==` is really `identical?` and I'm guessing that the `getAsString()` call is returning an equivalent-but-different-object string. 🤦